common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -195,3 +198,168 @@ def test_revoke_unenroll_entitlement(self):
+        self.patcher = patch("openedx.core.djangoapps.catalog.utils.get_course_runs_for_course")
+        self.mock_get_course_runs = self.patcher.start()
+        self.mock_get_course_runs.return_value = [
+            {'key': str(self.course.id)},
+            {'key': str(self.course2.id)}
+        ]
+
+    def tearDown(self):
+        self.patcher.stop()
+        super(EntitlementEnrollmentViewSetTest, self).tearDown()
+
+    def test_user_can_enroll(self):
+        course_entitlement = CourseEntitlementFactory(
+            user=self.user
+        )
+        url = reverse(
+            self.ENTITLEMENTS_ENROLLMENT_NAMESPACE,
+            args=[str(course_entitlement.uuid)]
+        )
+        assert course_entitlement.enrollment_course_run is None
+
+        data = {
+            'course_run_id': str(self.course.id)
+        }
+        response = self.client.post(
+            url,
+            data=json.dumps(data),
+            content_type='application/json',
+        )
+        course_entitlement.refresh_from_db()  # pylint: disable=no-member
+
+        assert response.status_code == 201
+        assert CourseEnrollment.is_enrolled(self.user, self.course.id)
+        assert course_entitlement.enrollment_course_run is not None
+
+    def test_user_can_unenroll(self):
+        course_entitlement = CourseEntitlementFactory.create(user=self.user)
+        url = reverse(
+            self.ENTITLEMENTS_ENROLLMENT_NAMESPACE,
+            args=[str(course_entitlement.uuid)]
+        )
+        assert course_entitlement.enrollment_course_run is None
+
+        data = {
+            'course_run_id': str(self.course.id)
+        }
+        response = self.client.post(
+            url,
+            data=json.dumps(data),
+            content_type='application/json',
+        )
+        course_entitlement.refresh_from_db()
+
+        assert response.status_code == 201
+        assert CourseEnrollment.is_enrolled(self.user, self.course.id)
+
+        response = self.client.delete(
+            url,
+            content_type='application/json',
+        )
+        assert response.status_code == 204
+
+        course_entitlement.refresh_from_db()
+        assert not CourseEnrollment.is_enrolled(self.user, self.course.id)
+        assert course_entitlement.enrollment_course_run is None
+
+    def test_user_can_switch(self):
+
+        course_entitlement = CourseEntitlementFactory.create(user=self.user)
+
+        url = reverse(
+            self.ENTITLEMENTS_ENROLLMENT_NAMESPACE,
+            args=[str(course_entitlement.uuid)]
+        )
+        assert course_entitlement.enrollment_course_run is None
+
+        data = {
+            'course_run_id': str(self.course.id)
+        }
+        response = self.client.post(
+            url,
+            data=json.dumps(data),
+            content_type='application/json',
+        )
+        course_entitlement.refresh_from_db()
+
+        assert response.status_code == 201
```

Comment:
Just checking that this is intended, do we use 201 here for updates as well as creates?

Reply:
Yes, that is the intention. 201 always for a successful Enroll.